v8.5.6: refresh TopSQL docs and screenshots (#22559) (#22734)
pingcap/docs: dashboard/top-sql.md

---
title: TiDB Dashboard Top SQL page
summary: Use Top SQL to identify queries that consume the most CPU, network, and logical IO resources
---

# TiDB Dashboard Top SQL Page

On the Top SQL page of TiDB Dashboard, you can view and analyze the most resource-consuming SQL queries on a specified TiDB or TiKV node over a period of time.

- After you enable Top SQL, this feature continuously collects CPU workload data from existing TiDB and TiKV nodes and retains the data for up to 30 days.

> - Starting from v8.5.6 and v9.0.0, you can also enable **TiKV Network IO collection (multi-dimensional)** in the Top SQL settings to further view metrics such as `Network Bytes` and `Logical IO Bytes` for specified TiKV nodes, and perform aggregation analysis in dimensions of `By Query`, `By Table`, `By DB`, and `By Region`.

Reviewer: ```suggestion
- Starting from v8.5.6, you can also enable **TiKV Network IO collection (multi-dimensional)** in the Top SQL settings to further view metrics such as `Network Bytes` and `Logical IO Bytes` for specified TiKV nodes, and perform aggregation analysis in dimensions of `By Query`, `By Table`, `By DB`, and `By Region`.
```

SET GLOBAL tidb_enable_top_sql = 1;
```

> ### (Optional) Enable TiKV Network IO collection <span class="version-mark">New in v8.5.6 and v9.0.0</span>

Reviewer: ```suggestion
### (Optional) Enable TiKV Network IO collection <span class="version-mark">New in v8.5.6</span>
```

    If you are not sure which node to observe, you can first locate the node with abnormal workload from Grafana or the [TiDB Dashboard Overview page](/dashboard/dashboard-overview.md), and then return to the Top SQL page for further analysis.

3. Set the time range and refresh data as needed.

    You can adjust the time range in the time picker or zoom the observation window by selecting a time range in the chart. Setting a smaller time range displays more fine-grained data, with a precision of up to 1 second.

    ![Change time range](/media/dashboard/v8.5-top-sql-usage-change-timerange.png)

    If the chart is out of date, click **Refresh** to refresh once, or select the data auto-refresh frequency from the **Refresh** drop-down list.

    ![Refresh](/media/dashboard/v8.5-top-sql-usage-refresh.png)

4. Select the observation mode.

    - Use `Limit` to display the Top `5`, `20`, or `100` SQL queries.
    - The default aggregation dimension is `By Query`. If you select a TiKV node, you can also aggregate in dimensions of `By Table`, `By DB`, or `By Region`.

        ![Select aggregation dimension](/media/dashboard/v8.5-top-sql-usage-select-agg-by.png)

>     - The default sort order is `Order By CPU` (sorted by CPU time). If you select a TiKV node and have [enabled TiKV Network IO collection (multi-dimensional)](#optional-enable-tikv-network-io-collection-new-in-v856-and-v900), you can also select `Order By Network` (sorted by network bytes) or `Order By Logical IO` (sorted by logical IO bytes).

Reviewer: ```suggestion
    - The default sort order is `Order By CPU` (sorted by CPU time). If you select a TiKV node and have [enabled TiKV Network IO collection (multi-dimensional)](#optional-enable-tikv-network-io-collection-new-in-v856), you can also select `Order By Network` (sorted by network bytes) or `Order By Logical IO` (sorted by logical IO bytes).
```

        ![Select order by](/media/dashboard/v8.5-top-sql-usage-select-order-by.png)

    > **Note**
    >

>     > `By Region` and `Order By Network`, `Order By Logical IO` are only available when [TiKV Network IO collection (multi-dimensional)](#optional-enable-tikv-network-io-collection-new-in-v856-and-v900) is enabled. If this feature is not enabled but historical data still exists, the page will continue to display historical data and prompt that new data cannot be fully collected.

Reviewer: ```suggestion
    > `By Region`, `Order By Network`, and `Order By Logical IO` are only available when [TiKV Network IO collection (multi-dimensional)](#optional-enable-tikv-network-io-collection-new-in-v856) is enabled. If this feature is not enabled but historical data still exists, the page continues to display historical data and prompt that new data cannot be fully collected.
```
